lib/snap_and_clip_to_nhd.py

```diff
@@ -8,50 +8,40 @@
 from os.path import splitext
 from shapely.strtree import STRtree
 from shapely.geometry import Point,MultiLineString,LineString,mapping
-from derive_headwaters import findHeadWaterPoints
 
-def subset_vector_layers(hucCode,nwm_streams_fileName,nhd_streams_fileName,nhd_streams_vaa_fileName,nwm_lakes_fileName,nwm_catchments_fileName,wbd_fileName,subset_nhd_streams_fileName,subset_nwm_lakes_fileName,subset_nwm_headwaters_fileName,subset_nwm_catchments_fileName,subset_nwm_streams_fileName,subset_nhd_headwaters_fileName=None,dissolveLinks=False):
+def subset_vector_layers(hucCode,nwm_streams_fileName,nwm_headwaters_fileName,nhd_streams_fileName,nwm_lakes_fileName,nwm_catchments_fileName,wbd_fileName,wbd_buffer_fileName,subset_nhd_streams_fileName,subset_nwm_lakes_fileName,subset_nwm_headwaters_fileName,subset_nwm_catchments_fileName,subset_nwm_streams_fileName,subset_nhd_headwaters_fileName=None,dissolveLinks=False):
 
     hucUnitLength = len(str(hucCode))
 
     wbd = gpd.read_file(wbd_fileName)
+    wbd_buffer = gpd.read_file(wbd_buffer_fileName)
     projection = wbd.crs
 
     # find intersecting lakes and writeout
     print("Subsetting NWM Lakes for HUC{} {}".format(hucUnitLength,hucCode),flush=True)
-    nwm_lakes = gpd.read_file(nwm_lakes_fileName, mask = wbd)
+    nwm_lakes = gpd.read_file(nwm_lakes_fileName, mask = wbd_buffer)
     if not nwm_lakes.empty:
         nwm_lakes.to_file(subset_nwm_lakes_fileName,driver=getDriver(subset_nwm_lakes_fileName),index=False)
     del nwm_lakes
 
     # find intersecting nwm_catchments
     print("Subsetting NWM Catchments for HUC{} {}".format(hucUnitLength,hucCode),flush=True)
     nwm_catchments = gpd.read_file(nwm_catchments_fileName, mask = wbd)
```

Comment:
Why are we using a mask of wbd instead of wbd_buffer?

Reply:
I'm not sure if it would make much of a difference and we would need to retest. This would only apply headwater segments of adjacent catchments and incoming streams which already have a break at the confluence of the HUC of interest.